src/transformers/models/mamba/modeling_mamba.py

```diff
@@ -638,7 +638,12 @@ def prepare_inputs_for_generation(
     ):
         # Overwritten -- uses `cache_params` as opposed to `past_key_values`
 
+        cache_params_not_initialized = cache_params is None
         if use_cache:
+            if cache_params_not_initialized:
+                max_batch_size = inputs_embeds.size(0) if inputs_embeds is not None else input_ids.size(0)
+                cache_params = MambaCache(self.backbone.config, max_batch_size, device=self.device, dtype=self.dtype)
+                cache_position = torch.arange(0, self.backbone.config.conv_kernel, device=input_ids.device)
```

Comment:
why do we have changes in code? I'd rather separate in a different PR if this is for the init